integration-tests/tests/advanced-happy-path.spec.ts

```diff
@@ -0,0 +1,171 @@
+  componentInfo.set('goFileSHAUpdated', 'f395a5944d9e9c3ee88c348cb836a15af4115cea');
+  componentInfo.set('commitMessageUpdate', 'Changing the string value inside HelloServer func');
+  componentInfo.set('commitMessageRevert', 'Revert: Changing the string value inside HelloServer func');
+
+  after(function () {
+    // Revert the changes made to the file
+    latestCommitsTabPage.editFile(
+      publicRepos[0], componentInfo.get('filePath'),
+      componentInfo.get('commitMessageRevert'),
+      componentInfo.get('goFileBase64Original'),
+      componentInfo.get('goFileSHAUpdated')
+    );
+
+    Applications.deleteApplication(applicationName);
+  });
+
+  describe('Create an Application with a component', () => {
+    it('Set Application Name', () => {
+      Applications.createApplication(applicationName);
+    });
+
+    it('Add a component to Application', () => {
+      componentName = Common.generateAppName(componentName);
+
+      Applications.createComponent(publicRepos[0], componentName, true);
+      Applications.createdComponentExists(componentName, applicationName);
+      Applications.checkComponentStatus(componentName, 'Build Running');
+    });
+  });
+
+  describe('Trigger a new Pipelinerun related to push event', () => {
+    it('Merge the auto-generated PR', () => {
+      latestCommitsTabPage.mergePR(componentName, publicRepos[0]);
+    });
+
+    it('Record the commits on the PR', () => {
+      latestCommitsTabPage.fetchCommitSHAOnPR(componentName, publicRepos[0]);
+    });
+  });
+
+  describe('Explore "Pipeline runs" Tab', () => {
+    it("Verify the PipelineRuns List view", () => {
+      Applications.goToPipelinerunsTab();
+
+      cy.get('tbody').find("tr").then((row) => {
+        for (let i = 0; i < row.length; i++) {
+          cy.get('tbody tr').eq(i).then(($row) => {
+            cy.wrap($row).find('td').eq(0).find('a').then((pipelinerunName) => {
+              Applications.createdPipelinerunSucceeded(pipelinerunName.text().trim());
+              pipelinerunsTab.pipelineRunList.push(pipelinerunName.text().trim());
+            });
+          });
+        }
+      });
+    });
+  });
+
+  describe('Explore "Latest commits" Tab', () => {
+    it("Verify the Commits List view", () => {
+      Applications.goToLatestCommitsTab();
+
+      for (let commitObject of latestCommitsTabPage.commitList) {
+        Common.checkRowValues(commitObject.sha, [commitObject.title, "main", componentName, commitObject.user, "Succeeded"]);
+      }
+    });
+  });
+
+  describe('Check Component Deployment', () => {
+    it("Verify the status code and response body of the deployment URL of each component", () => {
+      Applications.goToComponentsTab();
+
+      applicationDetailPage.expandDetails(componentName);
+
+      cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+        cy.wait(20000);
```

Comment:
what are we waiting for here and why 20 seconds?

Reply:
done, thanks